packages/mui-utils/src/useId/useId.ts

```diff
@@ -31,6 +31,7 @@ export default function useId(idOverride?: string): string | undefined {
     const reactId = maybeReactUseId();
     return idOverride ?? reactId;
   }
+  // TODO: uncomment once we enable eslint-plugin-react-compiler // eslint-disable-next-line react-compiler/react-compiler -- useGlobalId is required outside the component
```

Comment:
This explanation is incorrect. Also, we can remove it because it's explained on the next line: "`React.useId` is invariant at runtime."

```suggestion
  // TODO: uncomment once we enable eslint-plugin-react-compiler // eslint-disable-next-line react-compiler/react-compiler
```